pinot-common/src/main/java/org/apache/pinot/common/utils/http/HttpClient.java

```diff
@@ -86,22 +87,23 @@ public class HttpClient implements AutoCloseable {
   private final CloseableHttpClient _httpClient;
 
   public HttpClient() {
-    this(null);
+    this(HttpClientConfig.DEFAULT_HTTP_CLIENT_CONFIG, null);
   }
 
-  public HttpClient(@Nullable SSLContext sslContext) {
+  public HttpClient(HttpClientConfig httpClientConfig, @Nullable SSLContext sslContext) {
```

Comment:
This is a change that breaks backward compatibility. We should add another constructor with only `SSLContext` as parameter

Reply:
How so? Is anyone extending this Class and using their own HttpClient implementations?

Reply:
Raised: https://github.com/apache/pinot/pull/10519